src/plugins/vis_type_timeseries/public/application/components/index_pattern.js

```diff
@@ -157,18 +172,46 @@ export const IndexPattern = ({ fields, prefix, onChange, disabled, model: _model
           </EuiFlexItem>
         </EuiFlexGroup>
       )}
+      {!!dataSourceManagementEnabled && (
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <EuiFormRow
+              id={htmlId('dataSource')}
+              label={i18n.translate('visTypeTimeseries.indexPattern.dataSourceLabel', {
+                defaultMessage: 'Data source',
+              })}
+            >
+              <DataSourceSelector
+                savedObjectsClient={getSavedObjectsClient().client}
+                notifications={getNotifications().toasts}
+                onSelectedDataSource={handleDataSourceSelectChange}
+                defaultOption={model.data_source_id ? [{ id: model.data_source_id }] : [{ id: '' }]}
+                disabled={false}
+                fullWidth={false}
+                removePrepend={true}
+                isClearable={false}
+                hideLocalCluster={!!getHideLocalCluster().hideLocalCluster}
```

Comment:
I don't think we need to set `hideLocalCluster` any more. As #6361 merged in, this prop will be decided by the data source management plugin. Whatever get passed in explicitly by the props, will be overwritten anyways.

Reply:
Removed all instances of `hideLocalCluster`